File: app/src/main/java/com/vikingsen/cheesedemo/inject/Injector.java
package com.vikingsen.cheesedemo.inject;


import android.app.Application;

public class Injector {
    private static AppComponent appComponent = null;

    public static void init(Application application) {
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(application))
                .build();
    }

    public static AppComponent get() {
        if (appComponent == null) {
            throw new RuntimeException("Must call init before calling get.");
        }
        return appComponent;
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/inject/Injector.kt
package com.vikingsen.cheesedemo.inject


import android.app.Application

object Injector {
    private lateinit var appComponent: AppComponent

    @JvmStatic
    @Synchronized
    fun init(application: Application) {
        appComponent = DaggerAppComponent.builder()
                .appModule(AppModule(application))
                .build()
    }

    @JvmStatic
    @Synchronized
    fun get(): AppComponent = appComponent
}
